src/compile-middleware.ts

```diff
@@ -84,56 +104,195 @@ export function babelCompile(forceCompile: boolean): RequestHandler {
-      console.warn(`Error compiling script in ${location}: ${e.message}`);
+  let requireScriptTag, wctScriptTag;
+
+  for (const scriptTag of dom5.queryAll(document, isJsScriptNode)) {
+    // Is this a module script we should transform?
+    const transformingModule = options.transformModules &&
+        dom5.getAttribute(scriptTag, 'type') === 'module';
+
+    if (transformingModule && !requireScriptTag) {
+      // We need RequireJS to load the AMD modules we are declaring. Inject the
+      // dependency as late as possible (right before the first module is
+      // declared) because some of our legacy non-module dependencies,
+      // typically loaded in <head>, behave differently when window.require is
+      // present.
+      const fragment = parse5.parseFragment(
+          '<script src="/components/requirejs/require.js"></script>\n');
+      requireScriptTag = fragment.childNodes[0];
+      dom5.insertBefore(scriptTag.parentNode, scriptTag, fragment);
+    }
+
+    const src = dom5.getAttribute(scriptTag, 'src');
+    const isInline = !src;
+
+    if (src && src.includes('web-component-tester/browser.js')) {
+      wctScriptTag = scriptTag;
+    }
+
+    if (transformingModule && !isInline) {
+      // Transform an external module script into a `require` for that module,
+      // to be executed immediately.
+      dom5.replace(
+          scriptTag,
+          parse5.parseFragment(`<script>require(["${src}"]);</script>\n`));
+
+    } else if (isInline) {
+      let js = dom5.getTextContent(scriptTag);
+      const plugins = [];
+      if (options.transformES2015) {
+        plugins.push(...es2015Plugins);
+      }
+      if (transformingModule) {
+        plugins.push(...modulesPlugins);
+      }
+
+      let compiled;
+      try {
+        compiled = babelCore.transform(js, {plugins}).code;
+      } catch (e) {
+        // Leave the original script as-is. Maybe it will work...
+        // TODO Show an error in the browser console, or on the runner page.
+        console.warn(`Error compiling script in ${location}: ${e.message}`);
+        continue;
+      }
+
+      if (transformingModule) {
+        // The Babel AMD transformer output always starts with a `define` call,
+        // which registers a module but does not execute it immediately. Since
+        // we're in HTML, these are our top-level scripts, and we want them to
+        // execute immediately. Swap it out for `require` so that it does.
+        compiled = compiled.replace('define', 'require');
+
+        // Remove type="module" since this is non-module JavaScript now.
+        dom5.removeAttribute(scriptTag, 'type');
+      }
+
+      dom5.setTextContent(scriptTag, compiled);
     }
   }
+
+  if (wctScriptTag && requireScriptTag) {
+    // This looks like a Web Component Tester script, and we have converted
+    // ES modules to AMD. Converting a module to AMD means that it will not
+    // execute synchronously like an ES module would. Since WCT listens for the
```

Comment:
"synchronously" isn't quite the right word here. DOMCOntentLoaded waits for modules to run is all.